### DISCUSSION TITLE
Polymorphic Spotlight.Action component (for "a" tag or other link)

### DISCUSSION THREAD

**Discussion by QuinnStraus** [mantinedev/mantine#5404] [Feature requests]
I have been using a Spotlight in my project, but I was searching for a way to make the action buttons links.
I have been hacking it with a navigation passed in through the onClick on the actions, but this means I don't get nice features of links such as cmd+click opening in a new tab. 
I tried to add this through the Spotlight.Action, but since the component is not polymorphic, adding component="a" (or in my case the Next.js Link component) gave a type error when I tried to add the href prop (although it did work if I ignored the error). In addition, adding a link as the child means I have an "a" tag nested inside a button which violates the HTML specifications, and I lose some built-in spotlight action functionality (like highlighting).

Links inside the Spotlight seem like a common use case, so it would be nice to have built-in support at least on the Spotlight.Action component, and perhaps on the parent Spotlight component as well - although I can see it being a challenge to set up the types correctly there.

**Comment by ismailenesayhan** [comment]
I solved this problem using react-router-dom library. Maybe it helps

``` js
// data
const SpotLightData = [
    {
      id: "1",
      slug: "page-slug",
      title: "Page Title",
      description: "Page Description",
    },
    {
      id: "2",
      slug: "page-slug2",
      title: "Page Title2",
      description: "Page Description2",
    },
  ];

// map
  const actions = SpotLightData?.map((item) => ({
    id: item.id,
    label: item.title,
    description: item.description,
    onClick: () => navigate(`/${item.slug}`),
  }));

// component
 <Spotlight actions={actions || []} />
```

**Reply by QuinnStraus** [reply]
I am doing a similar thing on my end and it does work to navigate. I just want the added functionality of having the buttons be actual links rather than just buttons that navigate so I can do things like cmd + click to open in a new tab.

**Comment by krishna-santosh** [comment]
+1

**Comment by gabekus** [comment]
+1

**Comment by willpinha** [comment]
I am using [window.location.assign](https://developer.mozilla.org/docs/Web/API/Location/assign) to simulate an anchor tag (`<a>`). Note that this approach does not use client-side navigation, and therefore reloads the entire page. If you want client-side navigation, use a routing library like React Router

```ts
const actions: SpotlightActionData[] = [
  {
    id: "a",
    label: "A",
    onClick: () => window.location.assign("/a");
];
```

**Comment by chrisBas** [comment]
not that this is an ideal solution, but being it currently isn't fully polymorphic, you can hack your way through it with something like this:

```
config = {href: "/dashboard/1"}
<Spotlight.Action
     component={CustomAnchor}
     // this is needed b/c the elements are not truely polymorphic, so to pass an href we can do this
     {...config} 
/>
     
const CustomAnchor = ({href,onTrigger, ...props}, {href: string}) => {
     return <Link {...props} href={href}  />
}

```

NOTE: IFF you need to pass onClick, it must be passed with care, one of the following approaches should help depending on the usecase:
1. be passed directly to Spotlight.Action so that Spotlight.Action can wrap it with additional logic (such as closing the Spotlight modal)
2. specify an onClick in CustomAnchor as something like onClick={(e) => { (props as any)?.onClick(e); // additional logic} }
